src/lib.rs

```diff
@@ -221,6 +235,51 @@ macro_rules! get_function {
 get_function!(name: get_mut, mutability: mut);
 get_function!(name: get, mutability: );
 
+/// Macro to parametrise over mutability in get_node methods.
+macro_rules! get_node_function {
+    (
+        name: $name:ident,
+        mutability: $($mut_:tt)*
+    ) => {
+        id!(fn $name<'a, K, V>(
+            trie: &'a $($mut_)* Trie<K, V>,
+            mut key_fragments: NibbleVec
+        ) -> Option<&'a $($mut_)* Trie<K, V> > where K: TrieKey {
+            // Handle retrieval at the root.
+            if key_fragments.len() == 0 {
+                println!("hello");
+                return Some(trie);
+            }
+
+            let bucket = key_fragments.get(0) as usize;
+
+            match trie.children[bucket] {
+                None => None,
+                Some(ref $($mut_)* existing_child) => {
+                    match match_keys(&key_fragments, &existing_child.key) {
+                        KeyMatch::Full => Some(existing_child),
+                        KeyMatch::Partial(idx) => {
+                            let new_key_fragments = key_fragments.split(idx);
+                            $name(& $($mut_)* *existing_child, new_key_fragments)
+                        },
+                        KeyMatch::FirstPrefix => Some(existing_child),
```

Comment:
hmmm i need to check the internal, because the string can be very long (full books) , so I need to make sure the size of the input string does not influence the time it takes for get_ancestor_value to return me the same result.

After my 2 other concerns , which are the reason why i didn't use get_ancestor_value  was that

  1. once i got 首次将 , then  i need to work  on  首次将有女性成为美元纸币的正面人物  etc.  (but I guess this it's just me not being yet confident with slice) 
  2. i also got the case on words not in the dictionnary like for example  `首次将有Obama女性成为美元纸币的正面人物` for which my algorithm add characters until i met a character that start to make sense again for me

so currently my algorithm is working on iteration of `myString.chars().fold(...)` and I wonder how it will integrate with get_ancestor value